miniwin/sdl3gpu/src/miniwin_d3drmviewport.cpp

```diff
@@ -61,25 +61,19 @@ static void D3DRMMatrixInvertOrthogonal(D3DRMMATRIX4D out, const D3DRMMATRIX4D m
 	out[3][2] = -(out[0][2] * t.x + out[1][2] * t.y + out[2][2] * t.z);
 }
 
-static void HMM_Perspective_LH_NO(D3DRMMATRIX4D Result, float FOV, float AspectRatio, float Near, float Far)
+static void CalculateProjectionMatrix(D3DRMMATRIX4D Result, float field, float aspect, float near, float far)
 {
-	for (int i = 0; i < 4; i++) {
-		std::fill(Result[i], Result[i] + 4, 0.f);
-	}
-
-	// See https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/gluPerspective.xml
-
-	float Cotangent = 1.0f / SDL_tanf(FOV / 2.0f);
-	Result[0][0] = Cotangent / AspectRatio;
-	Result[1][1] = Cotangent;
-	Result[2][3] = -1.0f;
-
-	Result[2][2] = (Near + Far) / (Near - Far);
-	Result[3][2] = (2.0f * Near * Far) / (Near - Far);
+	float f = near / field;
+	float debth = far - near;
+
+	D3DRMMATRIX4D perspective = {
+		{f, 0, 0, 0},
+		{0, f * aspect, 0, 0},
+		{0, 0, far / debth, 1},
+		{0, 0, (-near * far) / debth, 0},
```

Comment:
Assuming you meant depth
```suggestion
	float depth = far - near;

	D3DRMMATRIX4D perspective = {
		{f, 0, 0, 0},
		{0, f * aspect, 0, 0},
		{0, 0, far / depth, 1},
		{0, 0, (-near * far) / depth, 0},
```

Reply:
Thanks spelling is not my strong suit :)